### ISSUE TITLE
bootstrap code fails on some systems due to HTTP 302

### ISSUE THREAD

**Issue by aspiers** [quelpa/quelpa#194]
https://github.com/quelpa/quelpa#installation says:

> If you want to bootstrap quelpa automatically from your init file, you can use this snippet:
>
> ```
> (unless (package-installed-p 'quelpa)
>     (with-temp-buffer
>       (url-insert-file-contents "https://github.com/quelpa/quelpa/raw/master/quelpa.el")
>       (eval-buffer)
>       (quelpa-self-upgrade)))
> ```

However that URL returns an HTTP 302, so `url-insert-file-contents` inserts 0 bytes into the temporary buffer, and the self-upgrade fails.  In order to avoid a redirect, the correct URL should be https://raw.githubusercontent.com/quelpa/quelpa/master/quelpa.el

**Comment by steckerhalter** [comment]
Hi, it should work with the redirect. I suspect another problem. It might be that your request is blocked by github as we had here for example: https://github.com/quelpa/quelpa/issues/191

**Comment by aspiers** [comment]
I'm AFK now but I'm pretty sure replacing the URL fixed it. However it worked on a few machines with 26.3 but not another with 26.1.

**Comment by steckerhalter** [comment]
why should the redirect be a problem? `url-insert-file-contents` should work with redirects AFAIK. if it returns empty content there is probably another issue unrelated to the redirect.

**Comment by aspiers** [comment]
It might be a problem if 26.1 doesn't handle them correctly.  Like I said, I'm pretty sure replacing the URL with the non-redirected one fixed it, so that would be a clear indication that the issue is the 302.

**Comment by steckerhalter** [comment]
i changed the address: https://github.com/quelpa/quelpa/commit/f1fc228f217be692eaae2d53f51966ce922d6a32

**Comment by aspiers** [comment]
Thanks!
